### PR TITLE
[Unittest] Add network_graph & lr_cosine unit test case

### PR DESCRIPTION
**Pull request by DonghakPark**
1. [[Unittest] Add network_graph unit test](https://github.com/nnstreamer/nntrainer/pull/2762/commits/72ecd76f69911a4b7dcb64900188f15069031edb)
- add reinitialize / catch exceptions
- These tests are not limited to specific layers but rather serve as test cases for checking activities performed on actual network graphs.

2. [[Unittest] Add lr_scheduler_cosine unit test](https://github.com/nnstreamer/nntrainer/pull/2762/commits/5423dabe7d9d9cb50b8532f6e262c0d0257adb37)
- property setting test / finalize test / getLearningRate test
- add to api and make helper function 

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

**Comment by DonghakPark** [comment]
After adding more diverse test cases for functions and classes, I will remove the "Do Not Merge" label and Request review.

**Review by baek2sm — APPROVED** [review]
LGTM !!
I like it because there are many negative test cases :)

**Review by skykongkong8 — APPROVED** [review]
LGTM
